batavia/types/Dict.js

```diff
@@ -52,19 +52,30 @@ batavia.types.Dict = function() {
      **************************************************/
 
     Dict.prototype.__lt__ = function(other) {
-        return this.valueOf() < other;
+        throw new batavia.builtins.TypeError("unorderable types: dict() < dict()");
     };
 
     Dict.prototype.__le__ = function(other) {
-        return this.valueOf() <= other;
+        throw new batavia.builtins.TypeError("unorderable types: dict() <= dict()");
     };
 
     Dict.prototype.__eq__ = function(other) {
-        return this.valueOf() == other;
+        if (batavia.isinstance(other, [batavia.types.NoneType]) == true) return false;
```

Comment:
![robert in15 dict_window](https://cloud.githubusercontent.com/assets/19719405/15809209/cc81293a-2b40-11e6-90a8-2153d45d9307.png)

Pythonic "==" appears to throw no errors when dict is involved.  I can change it regardless, if I receive confirmation.